### PR TITLE
UIK-31: Create User Model

### PR DESCRIPTION
**Pull request by StankovskiA**
This PR creates the user model, implements JWT authentication along with Register, Login & Logout endpoints. Additionally, adds a get user endpoint which returns the user.

### CODE REVIEW

File: furryfeet_be/furryfeet/views.py [hunk 1]
```diff
@@ -1,14 +1,80 @@
...
+        user = User.objects.filter(email=email).first()
+        if user is None:
+            raise AuthenticationFailed("User not found!")
+        
+        if not user.check_password(password):
+            raise AuthenticationFailed("Incorrect password!")
+
+        payload = {
+            'id': user.id,
+            'exp': datetime.utcnow() + timedelta(minutes=120),
```
Comment (SaraMisajlovska) [inline furryfeet_be/furryfeet/views.py:41]: ova e za token expiry taka?

Comment (StankovskiA) [reply furryfeet_be/furryfeet/views.py:41]: yessir

File: furryfeet_be/furryfeet/views.py [hunk 2]
```diff
@@ -1,14 +1,80 @@
...
+    def post(self, request):
+        serializer = UserSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        
+        return Response(serializer.data)
+    
+class LoginView(APIView):
+    def post(self, request):
+        email = request.data['email']
```
Comment (SaraMisajlovska) [inline furryfeet_be/furryfeet/views.py:29]: ako ne koristime username za login togas moze i ne ni treba kako property u modelot? ili za neso drugo ke ni treba?

Comment (StankovskiA) [reply furryfeet_be/furryfeet/views.py:29]: mhm, vo modelot e napraveno email da se koristi kako username

Comment (SaraMisajlovska) [reply furryfeet_be/furryfeet/views.py:29]: epa mozes da go trgnes toas i guess. poso i na FE nema da im treba za niso

**Review by SaraMisajlovska — APPROVED** [review]
great job !

**Review by MuratUseini123 — APPROVED** [review]

**Pull request merged**